test-harness.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+START=$(date "+%s")
+
+set -e
+
+ENV_FILE=".test-env"
+source $ENV_FILE
+
+echo "test-harness.sh: AFTER sourcing from $ENV_FILE. Build TYPE=$TYPE"
+
+rootdir=$(dirname "$0")
+pushd "$rootdir"
+
+## Reset test harness
+if [ -d "$SDK_TESTING_HARNESS" ]; then
+  pushd "$SDK_TESTING_HARNESS"
+  ./scripts/down.sh
+  popd
+  rm -rf "$SDK_TESTING_HARNESS"
+else
+  echo "test-harness.sh: directory $SDK_TESTING_HARNESS does not exist - NOOP"
+fi
+
+git clone --single-branch --branch "$SDK_TESTING_BRANCH" "$SDK_TESTING_URL" "$SDK_TESTING_HARNESS"
+
+## OVERWRITE incoming .env with .test-env
+cp "$ENV_FILE" "$SDK_TESTING_HARNESS"/.env
```

Comment:
@tzaffi Sorry - During our earlier verbal, I misunderstood the relationship between `.test-env` and algorand-sdk-testing `.env`.  

I thought we had override if-absent capabilities.  I see several options:
* Implement a merge giving preference to algorand-sdk-testing.  Allows continued use of a single environment file.
  * At least 2 options exists:  https://stackoverflow.com/a/14017638 and https://stackoverflow.com/a/57691790. 
  * I tested the former and I'm convinced it produces a correct diff.  If used, I'd cite the link source because it'd otherwise take me some time to decipher usage.
* Partition `SDK_TESTING_*` into a separate environment file.  Avoids the merge at the expense of 2nd config file.

Based on the PR's spirit, it seems the merge option is preferred.

Reply:
Additionally, assuming merge happens, then following line must be moved from earlier in the script to after the highlighted line.

```
echo "test-harness.sh: AFTER sourcing from $ENV_FILE. Build TYPE=$TYPE"
```

In its current location, `$TYPE` won't be populated.

Reply:
more experimentation

Reply:
A note I missed previously:  At least with the `awk` approach, assuming no further customization, in-place changes are unsupported.  So, I think you'll need a 3rd temporary file like:

```
awk -F= '!a[$1]++' "$ENV_FILE" "$SDK_TESTING_HARNESS"/.env.tmp
mv "$SDK_TESTING_HARNESS"/.env "$SDK_TESTING_HARNESS"/.env.tmp
```

Reply:
I hope it's okay with you. I've opted for a "KISS" approach. As I explained in the previous comment, we don't overwrite testing's .env's unless `OVERWRITE_TESTING_ENVIRONMENT=1`, but in that case, the onus is left to the developer to explicitly provide out all of the expected env vars. LMK if you are of the opinion that a more sophisticated approach as suggested above is still warranted.

Reply:
@tzaffi Resolving - The overwrite flag works for me.  Thanks for working through the feedback to find a path forward.

Aside:  Here's a briefly considered thought for a future iteration.
* If we add merge support, it'd simplify promoting the Cucumber test tags into algorand-sdk-testing.  I think the idea reduces the chance of an SDK failing to declare support for a set of tests.
* In the case where an SDK lags behind, the SDK _only_ overrides _only_ a single environment variable to declare its supported tests.
* I think there's still friction, but in my imagination, it'd be a net improvement over the status quo.